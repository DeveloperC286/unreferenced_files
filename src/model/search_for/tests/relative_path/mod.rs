use super::*;

const SEARCH_FOR_RELATIVE_PATH: bool = true;
const SEARCH_FOR_FILE_NAME: bool = false;
const SEARCH_FOR_FILE_STEM: bool = false;

#[test]
fn test_relative_path_searching_for_found() {
    // Given
    let search_for = UNREFERENCED_FILE1.clone();
    let searching = raw_file_with_content!(
        "@test\npublic void testImporting() {\n  import(\"./folder/file1.txt\");\n}"
    );

    // When
    let unreferenced_files = search_for.get_unreferenced_files(
        searching,
        SEARCH_FOR_RELATIVE_PATH,
        SEARCH_FOR_FILE_NAME,
        SEARCH_FOR_FILE_STEM,
    );

    // Then
    assert_unreferenced_files_snapshot!(FOUND, unreferenced_files);
}

#[test]
fn test_relative_path_searching_for_not_found() {
    // Given
    let search_for = UNREFERENCED_FILE1.clone();
    let searching =
        raw_file_with_content!("@test\npublic void testImporting() {\n  import(\"file1.txt\");\n}");

    // When
    let unreferenced_files = search_for.get_unreferenced_files(
        searching,
        SEARCH_FOR_RELATIVE_PATH,
        SEARCH_FOR_FILE_NAME,
        SEARCH_FOR_FILE_STEM,
    );

    // Then
    assert_unreferenced_files_snapshot!(NOT_FOUND, unreferenced_files);
}
