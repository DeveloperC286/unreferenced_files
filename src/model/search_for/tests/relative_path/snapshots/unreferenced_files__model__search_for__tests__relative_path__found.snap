---
source: src/model/search_for/tests/relative_path/mod.rs
expression: sorted_search_for

---
[]
